pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -337,6 +337,23 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 				// risk is that when the controller is now restarted before the field is set, another TaskRun will be created
 				ctxlog.Error(ctx, err, "Failed to update BuildRun status is ignored", namespace, request.Namespace, name, request.Name)
 			}
+
+			// Increase BuildRun count in metrics
+			buildmetrics.BuildRunCountInc(buildRun.Status.BuildSpec.StrategyRef.Name)
+
+			// Report the buildrun established duration (time between the creation of the buildrun and the start of the buildrun)
+			buildmetrics.BuildRunEstablishObserve(
+				buildRun.Status.BuildSpec.StrategyRef.Name,
+				buildRun.Namespace,
+				buildRun.Status.StartTime.Time.Sub(buildRun.CreationTimestamp.Time),
+			)
```

Comment:
I do not think that this one can work here because the buildRun does not yet have a StartTime. This is set later when the TaskRun started.

Reply:
Let me double check.

Reply:
I have no idea what I did in my local test with `minikube` that did not show up. Today, I was able to reproduce that this leads to an expected nil pointer de-reference scenario. Therefore, I moved the code section to the part of the code where the required start time is actually set for the first time.